### ISSUE TITLE
Feature Request: Add support for /etc/ethertypes

### ISSUE THREAD

**Issue by sapslaj** [TheDiveO/netdb#2]
I'm working on replacing an ad-hoc implementation of protocol and service lookups in one of my projects, [a NetFlow collection and metadata enrichment agent](https://github.com/sapslaj/morbius) with this library. It was pretty simple to slot this library in place for protocols and services, but my project also supports adding some metadata about the Ethernet protocol type of the flow. Adding support for /etc/ethertypes to this project would be really helpful and I'd be able to get rid of the [hard-coded mappings](https://github.com/sapslaj/morbius/blob/c8dfb3e41a3db478e234e2b67319f01577d39fe0/enricher/protonames_enricher.go#L74-L104).

Some ideas for the API:

```go
func main() {
	ipv4 := netdb.EtherTypeByNumber(0x0800)
	fmt.Println(ipv4.Name)
	// Output: IPv4
	fmt.Println(ipv4.Number)
	// Output: 2048
	fmt.Println(ipv4.Aliases)
	// Output: [ip ipv4]
	fmt.Println(ipv4.Comment)
	// Output: Internet IP (IPv4)
}
```

```go
func main() {
	dot1q := netdb.EtherTypeByName("dot1q")
	fmt.Println(dot1q.Name)
	// Output: 802_1q
	fmt.Println(dot1q.Number)
	// Output: 33024
	fmt.Println(dot1q.Aliases)
	// Output: [8021q 1q 802.1q dot1q]
	fmt.Println(dot1q.Comment)
	// Output: 802.1Q Virtual LAN tagged frame
}
```

Not totally sure if the comment parsing is really necessary, but it does contain some useful info sometimes.

**Comment by thediveo** [comment]
Sounds good, even if Debian's `/etc/ethertypes` is a little bit thin. API looks good to me, it follows and should follow the pattern this module uses for protocols and services. I've checked with the upstream Debian repo to make sure it contains an ethertypes to auto-generate the builtin data from.

**Comment by sapslaj** [comment]
Interestingly Debian's `/etc/ethertypes` is more comprehensive than what ships with Arch Linux, at least based on what my machine has. Either way, it's pretty easy for someone to add in more entries to the index if needed so it's _probably_ fine.

~I got a WIP branch that implements this that I can share in just a bit.~

WIP branch here: https://github.com/sapslaj/netdb/tree/ethertypes-support

I'll make a proper PR once I get through some TODOs

**Comment by thediveo** [comment]
Ah, a PR sounds really great!

(I'm a little bit disappointed that Debian doesn't include the Profinet Ethertypes 😀)

**Comment by sapslaj** [comment]
Oh man does Profinet have its own EtherType? I've only ever dealt with EtherNet/IP and IIRC it uses regular TCP and UDP over IP (it's been a few years though so could be wrong...)

**Comment by thediveo** [comment]
released as v1.1.0; thank you very much!
